src/java/org/apache/cassandra/index/sai/plan/Operation.java

```diff
@@ -311,6 +311,12 @@ static Node buildExpression(RowFilter.Expression expression)
                     return new EmptyNode();
                 return node;
             }
+            else if (expression.operator() == Operator.ANALYZER_MATCHES && isDisjunction)
+            {
+                OperatorNode node = new AndNode();
+                node.add(new ExpressionNode(expression));
```

Comment:
can we add a comment ?
it is not straightforward to understand why we have to wrap the node with an AndNode with only one single child

Reply:
This PR is a cherry-pick from `main`. But your comment was acknowledged; it will be handled separately later. I will close the PR with more historical info, as this hotfix won't be needed.